mlflow/types/type_hints.py

```diff
@@ -262,21 +272,26 @@ def _get_element_type_of_list_type_hint(type_hint: type[list[Any]]) -> Any:
     Get the element type of list[...] type hint
     """
     args = get_args(type_hint)
+    # Optional[list[...]]
+    if type(None) in args:
+        raise MlflowException.invalid_parameter_value(OPTIONAL_INPUT_MSG)
     # a valid list[...] type hint must only contain one argument
     if len(args) == 0:
         raise MlflowException.invalid_parameter_value(
-            f"List type hint must contain the element type, got {type_hint}"
+            f"Type hint `{_type_hint_repr(type_hint)}` doesn't contain element type, fix it "
+            "by adding an element type, e.g. `list[str]` instead of `list`."
         )
     if len(args) > 1:
         raise MlflowException.invalid_parameter_value(
-            f"List type hint must contain only one element type, got {type_hint}"
+            f"Type hint `{_type_hint_repr(type_hint)}` contains {len(args)} element types, "
+            "only one element type is allowed."
```

Comment:
```suggestion
            f"Type hint `{_type_hint_repr(type_hint)}` contains {len(args)} element types. "
            "Collections must have only a single type definition e.g. `list[int]` is valid; `list[str, int]` is invalid."
```
Would it be possible to add a supplemental link to the docs on signature errors like this and others so that users can see direct guidance on these type definition errors? While it may seem repetitive to us to have this displayed all over the place, for a user they might only hit one of these issues when building their solution and ensuring that any edge cases can direct them to the docs for guidance is probably for the best. :)

Reply:
Yes I'm supposed to file a docs PR and then update all those error messages to link to the documentation page